sdk/lib/crt/misc/i10output.c

```diff
@@ -1,5 +1,140 @@
 #include <precomp.h>
 
+// From Wine msvcrt.h
+typedef struct {ULONG x80[3];} MSVCRT__LDOUBLE; /* Intel 80 bit FP format has sizeof() 12 */
+enum fpmod {
+    FP_ROUND_ZERO, /* only used when dropped part contains only zeros */
+    FP_ROUND_DOWN,
+    FP_ROUND_EVEN,
+    FP_ROUND_UP,
+    FP_VAL_INFINITY,
+    FP_VAL_NAN
+};
+struct fpnum {
+    int sign;
+    int exp;
+    ULONGLONG m;
+    enum fpmod mod;
+};
+
+// From wine bnum.h
+#define EXP_BITS 11
+#define MANT_BITS 53
+
+int fpnum_double(struct fpnum *fp, double *d)
+{
+    ULONGLONG bits = 0;
+
+    if (fp->mod == FP_VAL_INFINITY)
+    {
+        *d = fp->sign * INFINITY;
+        return 0;
+    }
+
+    if (fp->mod == FP_VAL_NAN)
+    {
+        bits = ~0;
+        if (fp->sign == 1)
+            bits &= ~((ULONGLONG)1 << (MANT_BITS + EXP_BITS - 1));
+        *d = *(double*)&bits;
+        return 0;
+    }
+
+    TRACE("%c %#I64x *2^%d (round %d)\n", fp->sign == -1 ? '-' : '+',
+        fp->m, fp->exp, fp->mod);
+    if (!fp->m)
+    {
+        *d = fp->sign * 0.0;
+        return 0;
+    }
+
+    /* make sure that we don't overflow modifying exponent */
+    if (fp->exp > 1<<EXP_BITS)
+    {
+        *d = fp->sign * INFINITY;
+        return ERANGE;
+    }
+    if (fp->exp < -(1<<EXP_BITS))
+    {
+        *d = fp->sign * 0.0;
+        return ERANGE;
+    }
+    fp->exp += MANT_BITS - 1;
+
+    /* normalize mantissa */
+    while(fp->m < (ULONGLONG)1 << (MANT_BITS-1))
```

Comment:
```suggestion
    while (fp->m < (ULONGLONG)1 << (MANT_BITS-1))
```
same for the two others below (lines 70 and 91).

Reply:
It's wine code, so I don't see any benefit in changing it, only to apply our coding standards. It's only going to make future syncs harder, or it's simply going to be lost.

Reply:
ah ok i didn't know.